api/v1alpha/api.proto

```diff
@@ -37,6 +37,114 @@ enum ImageType {
         IMAGE_TYPE_OCI       = 3;
 }
 
+// InsecureOption defines the insecure option flags to be used.
+enum InsecureOption {
+        INSECURE_OPTION_NONE    = 0;
+        INSECURE_OPTION_ALL     = 1;
+        INSECURE_OPTION_IMAGE   = 2;
+        INSECURE_OPTION_TLS     = 4;
+        INSECURE_OPTION_ON_DISK = 8;
+        INSECURE_OPTION_HTTP    = 16;
+        INSECURE_OPTION_PUBKEY  = 32;
+}
+
+// DiscoverOption defines the discovery options to be used for fetching.
+enum DiscoverOption {
```

Comment:
I'm not sure DiscoveryOptions is the right name (since it's not related to discovery but to the fetching behavior).

I'm also going to propose a little behavioral change on how fetch vs prepare/run should work respect on store usage since IMHO the current state is not completely clear to the users (and will be useful for OCI image caching). The basic idea is that fetch (as the names says) will always go to the net and import/update the blobs in the store, while prepare/run will just use what's in the store if avalable or fetch what's missing). But this deserves a broader discussion.